File: app/src/main/java/me/lukeforit/spaceofaday/ui/home/HomeViewModel.java
package me.lukeforit.spaceofaday.ui.home;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import javax.inject.Inject;

import me.lukeforit.spaceofaday.ui.base.Event;

public class HomeViewModel extends ViewModel {

    private MutableLiveData<Event<String>> openApodScreen = new MutableLiveData<>();

    @Inject
    public HomeViewModel() {
    }

    public void displayApod(String date) {
        openApodScreen.postValue(new Event<>(date));
    }

    public MutableLiveData<Event<String>> getOpenApodScreen() {
        return openApodScreen;
    }
}

File: app/src/main/java/me/lukeforit/spaceofaday/ui/home/HomeViewModel.kt
package me.lukeforit.spaceofaday.ui.home

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

import javax.inject.Inject

import me.lukeforit.spaceofaday.ui.base.Event

class HomeViewModel @Inject
constructor() : ViewModel() {

    val openApodScreen = MutableLiveData<Event<String>>()

    fun displayApod(date: String) {
        openApodScreen.postValue(Event(date))
    }
}
